plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeColumnProjectionInfo.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.ImmutableList;
+import io.airlift.slice.SizeOf;
+import io.trino.plugin.hive.HiveColumnProjectionInfo;
+import io.trino.spi.type.Type;
+
+import java.util.List;
+import java.util.Objects;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.airlift.slice.SizeOf.estimatedSizeOf;
+import static io.airlift.slice.SizeOf.instanceSize;
+import static io.trino.plugin.deltalake.DeltaHiveTypeTranslator.toHiveType;
+import static java.util.Objects.requireNonNull;
+
+public class DeltaLakeColumnProjectionInfo
+{
+    private static final int INSTANCE_SIZE = instanceSize(DeltaLakeColumnProjectionInfo.class);
+
+    private final Type type;
+    private final List<Integer> dereferenceIndices;
+    private final List<String> dereferenceNames;
+
+    @JsonCreator
+    public DeltaLakeColumnProjectionInfo(
+            @JsonProperty("type") Type type,
+            @JsonProperty("dereferenceIndices") List<Integer> dereferenceIndices,
+            @JsonProperty("dereferenceNames") List<String> dereferenceNames)
```

Comment:
currently, `dereferenceNames` will be the physical names of nested fields.

should I introduce `dereferenceNames` and `dereferencePhysicalNames`?

Reply:
CC: @findepi | @findinpath

Reply:
@krvikash yes, i think `dereferencePhysicalNames` is more appropriate for the purpose of this field.